discord/abc.py

```diff
@@ -1529,6 +1529,48 @@ async def pins(self) -> List[Message]:
         state = self._state
         data = await state.http.pins_from(channel.id)
         return [state.create_message(channel=channel, data=m) for m in data]
+    
+    def can_send(self, *objects) -> bool:
+        """Returns a :class:`bool` indicating whether you have the permissions to send the object(s)."""
+        mapping = {
+            'Message': 'send_messages',
+            'Embed': 'embed_links',
+            'File': 'attach_files',
+            'Emoji': 'use_external_emojis',
+            'GuildSticker': 'use_external_stickers',
+        }
+        # Can't use channel = await self._get_channel() since its async
+        if hasattr(self, 'permissions_for'):
+            channel = self
+        elif hasattr(self, 'channel'):
```

Comment:
It should be
```suggestion
        elif not type(self.channel).__name__ in ('DMChannel', 'GroupChannel')
```
instead because DMChannels have the channel attr

Reply:
Ah yes thanks